src/stores/store.ts

```diff
@@ -0,0 +1,160 @@
+import { defineStore, skipHydrate } from "pinia";
+import { spaceFilter } from "~/lib/filter-spaces";
+import type { Building, BuildingI18n } from "~/types/Building";
+import type { Filters } from "~/types/filters";
+import type { Space, SpaceI18n } from "~/types/Space";
+
+export const useStore = defineStore("index", () => {
+  const isMapMode = ref(false);
+  const isMobile = ref(false);
+
+  const selection = ref({
+    building: undefined as undefined | Building,
+    space: undefined as undefined | Space,
+  });
+
+  function clearSelection() {
+    selection.value = {
+      building: undefined,
+      space: undefined,
+    };
+  }
+
+  function selectBuilding(building: Building | undefined) {
+    selection.value = {
+      building,
+      space: undefined,
+    };
+  }
+
+  function selectSpace(space: Space | undefined) {
+    selection.value.space = space;
+  }
+
+  const defaultFilters: Filters = {
+    adjustableChairs: false,
+    buildings: [],
+    daylit: false,
+    ethernet: false,
+    studyType: [],
+    nearBathroom: false,
+    nearCoffeeMachine: false,
+    nearPrinter: false,
+    powerOutlets: false,
+    presentationScreen: false,
+    quietness: [],
+    showNearbyLocations: false,
+    showOpenLocations: false,
+    smartBoard: false,
+    stationaryPC: false,
+    whiteBoard: false,
+  };
+
+  const filters = useLocalStorage("filters", defaultFilters);
+
+  function clearFilters() {
+    filters.value = defaultFilters; //TODO: check if Vue3 reactivity works like that
+  }
+
+  const buildingsI18n = ref([] as BuildingI18n[]);
+
+  function setBuildings(buildings: BuildingI18n[]) {
+    buildingsI18n.value = buildings;
+  }
+
+  const locale = ref("en");
+
+  const buildings = computed(() => {
+    //const { locale } = useI18n();
+    return (
+      buildingsI18n.value
+        .map((buildingI18n) => {
+          const i18nProps = buildingI18n.i18n[locale.value];
+          return {
+            ...buildingI18n,
+            ...i18nProps,
+          } as Building;
+        })
+        //TODO: this fixes the behavior from prev version. OK?
+        .sort((buildingA, buildingB) =>
```

Comment:
Internals are hard to read. Can we put it into a named function that at least says how they are sorted?

Reply:
extracted a function.